homeassistant/components/simplisafe/__init__.py

```diff
@@ -523,27 +484,43 @@ async def update_system(system):
         for result in results:
             if isinstance(result, InvalidCredentialsError):
                 if self._emergency_refresh_token_used:
-                    _LOGGER.error(
-                        "SimpliSafe authentication disconnected. Please restart HASS"
+                    LOGGER.error(
+                        "Token disconnected or invalid. Please re-auth the "
+                        "SimpliSafe integration in HASS"
                     )
+
                     remove_listener = self._hass.data[DOMAIN][DATA_LISTENER].pop(
                         self._config_entry.entry_id
                     )
                     remove_listener()
```

Comment:
What happens now after we have update the config entry with new auth info? Should we still remove this listener? How will the integration continue to update?

Reply:
Good catch!